frontend/apps/db-explorer/src/client.ts

```diff
@@ -0,0 +1,166 @@
+import * as axios from 'axios';
+
+export type StringValue = { 'string': string; }
+
+export type Value =
+  StringValue
+  | { 'float': string }
+  | { int: string }
+  | { 'boolean': string }
+  | { 'null': boolean };
+
+export function isStringValue(value: Value): value is StringValue {
+  return (value as any)['string'] !== undefined;
+}
+
+export type Attributes = { [key: string]: Value };
+
+export type Record = {
+  id: string;
+  revision: string;
+  table: string;
+  attributes: Attributes;
+}
+
+export type RecordList = {
+  items: Record[];
+}
+
+export type GetRecordRequest = {
+  id: string;
+  table: string;
+  revision: string;
+}
+
+export type GetRecordsRequest = {
+  table: string;
+  recordIds?: string[];
+  revisions?: boolean;
+}
+
+export type PutRecordRequest = {
+  isReplication?: boolean;
```

Comment:
I'm a bit confused about the terms revision/replication? do you use them interchangeably or is there a difference in meaning?

Reply:
also, what problem was solved by adding this? why do we need to differentiate between original and copy? is this just a shortcut to not have to check for revisions every time?

Reply:
It's mostly due to (when we enable offline), that multiple people might edit the same records. When that happens, it's unclear whose version of the record is "the good one". To "solve" this, we store all the edits as different "revisions". Then, the engine chooses one of the revisions as being the "good one", but we can still present to the user that there was a conflict, and the user could resolve the conflict by choosing the "right one" manually. 

It's also needed for replication/synchronization between different devices. Basically, the devices would pull the revisions of the records that they don't know about. And since the process of "choosing the good one" is deterministic, all users will always elect the same revision all the time.

Reply:
In this case `isReplication` is exclusively used by the replicator. The engine assumes that the revision being `put` is known by another `engine`.